homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -54,6 +62,29 @@
 )
 
 
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config,
+    async_add_devices: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Import legacy configurations."""
+    _LOGGER.warning(
+        "Configuration in configuration.yaml is deprecated. Use the config flow instead."
+    )
+
+    @callback
+    def config_import(_):
+        """Import config entry when homeassistant has started."""
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+            )
+        )
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STARTED, config_import)
```

Comment:
Why do we wait for home assistant start?

Reply:
If I create the task directly without waiting for EVENT_HOMEASSISTANT_STARTED (EVENT_HOMEASSISTANT_START seems to be okay, too), the Config Entry stays in not loaded state until the next restart of HA. Is there a better way to handle this?

Reply:
I don't think that should be necessary. We'll have to look into this.

Reply:
Are you missing a dependency on another integration in the manifest? e.g., `ssdp`?

Reply:
I will investigate on this. Adding `ssdp` to the requirements did not fix it and from what I know, we do not have any additional requirements. In fact, even `ssdp` should not be required for the import.

Reply:
After some debugging, it does not make a difference, whether we add `ssdp` to the requirements or not. If there are no musiccast media players configured using the flow, the import works as expected without waiting for HA to be started. If there are already config entries for some musiccast media players, we run into the described problem (not leaded status) for those media players, which were newly imported. I have tried to debug this further and from what I can see, these new config entries from the import are generated at the same time as the already present musiccast config entries are loaded.

So if we just want to provide a migration strategy for those users, who use the current version of the integration, we could directly create the import task on platform setup. If we want to be able to support the configuration.yaml after some config entries are already present for musiccast in HA, we would have to investigate into that further or wait for home assistant to be started.

Reply:
The import is only meant as a migration for existing config YAML. We can add a check that we only import if there are no existing config entries.

Reply:
Okay, I implemented it that way